File: api/src/main/java/org/delivery/api/common/error/UserErrorCode.java
package org.delivery.api.common.error;

import lombok.AllArgsConstructor;
import lombok.Getter;
import org.springframework.http.HttpStatus;


/**
 * User의 경우 1000번대 에러코드 사용
 */
@AllArgsConstructor
@Getter
public enum UserErrorCode implements ErrorCodeIfs{

    // 외부, 내부, description
    //USER_NOT_FOUND(HttpStatus.NOT_FOUND.value(), 1484, "사용자를 찾을 수 없습니다"),
    USER_NOT_FOUND(HttpStatus.NOT_FOUND.value(), 1484, "일치 하는 회원 정보가 없습니다."),

    ;
    
    private final Integer httpStatusCode; // http status
    private final Integer errorCode; // internal(내부)코드
    private final String message;

}

File: common/src/main/kotlin/org/delivery/common/error/UserErrorCode.kt
package org.delivery.common.error

/**
 * User의 경우 1000번대 에러코드 사용
 */
enum class UserErrorCode(
    private val httpStatusCode: Int,
    private val errorCode: Int,
    private val message: String
): ErrorCodeIfs {
    // 외부, 내부, description
    //USER_NOT_FOUND(HttpStatus.NOT_FOUND.value(), 1484, "사용자를 찾을 수 없습니다"),
    USER_NOT_FOUND(404, 1484, "일치 하는 회원 정보가 없습니다."),
    ;

    override fun getHttpStatusCode(): Int {
        return this.httpStatusCode
    }

    override fun getErrorCode(): Int {
        return this.errorCode
    }

    override fun getMessage(): String {
        return this.message
    }
}